core/base-service/base.spec.js

```diff
@@ -101,27 +103,21 @@ describe('BaseService', function() {
       )
     })
 
-    it('Should throw if route is not overridden', async function() {
-      try {
-        await BaseService.invoke({}, {}, {})
-        expect.fail('Expected to throw')
-      } catch (e) {
-        expect(e.message).to.equal('Route not defined for BaseService')
-      }
+    it('Should throw if route is not overridden', function() {
+      return expect(BaseService.invoke({}, {}, {})).to.be.rejectedWith(
+        'Route not defined for BaseService'
```

Comment:
Should this use a regex instead? It seems to be doing partial string/contains validation (if you change the string message to just `BaseService` it will still pass)
```suggestion
        /^Route not defined for BaseService$/
```

Reply:
Good spot! I wasn't aware of this behavior. I fixed 3 other places using `throw()` or `rejectedWith()`.